src/lib/y2storage/blk_device.rb

```diff
@@ -305,10 +305,19 @@ def encrypted?
     # @param method [EncryptionMethod, Symbol] encryption method to create the new device
     # @param dm_name [String, nil] DeviceMapper table name of the new device
     # @param password [String, nil] password of the new device
+    # @param apqns [Array<EncryptionProcesses::Apqn>] APQNs to use when generating a secure key for
+    #   pervasive encryption.
+    #
     # @return [Encryption]
-    def encrypt(method: EncryptionMethod::LUKS1, dm_name: nil, password: nil)
+    def encrypt(method: EncryptionMethod::LUKS1, dm_name: nil, password: nil, apqns: [])
```

Comment:
np: So the `apqns` argument is only used depending on the `method` value. For the time being I guess it is OK, but in the future (if we get more specific options for each method) we could think about doing some refactoring (like having specific classes or methods for encryption method options. But, for the time being, let's keep it simple.